exasol/toolbox/nox/_documentation.py

```diff
@@ -88,6 +105,117 @@ def clean_docs(_session: Session) -> None:
         shutil.rmtree(docs_folder)
 
 
+@nox.session(name="docs:links", python=False)
+def docs_list_links(session: Session) -> None:
+    """List all the links within the documentation."""
+    ignore = [r".*"]
+    env = os.environ.copy()
+    env["SPHINX_EXTRA_LINKCHECK_IGNORES"] = ",".join(ignore)
+    with tempfile.TemporaryDirectory() as path:
+        tmpdir = Path(path)
+        sp = subprocess.run(
+            [
+                "poetry",
+                "run",
+                "--",
+                "sphinx-build",
+                "-b",
+                "linkcheck",
+                PROJECT_CONFIG.root / "doc",
+                tmpdir,
+            ],
+            capture_output=True,
+            text=True,
+            env=env,
+        )
+        print(sp.returncode)
+        if sp.returncode >= 2:
+            print(sp.stderr)
+            session.error(2)
+        output = tmpdir / "output.json"
+        links = output.read_text().split("\n")
+        file_links = []
+        for link in links:
+            if link != "":
+                line = json.loads(link)
+                if not line["uri"].startswith("#"):
+                    file_links.append(line)
+        file_links.sort(key=lambda file: file["filename"])
+        print(
+            "\n".join(
+                f"filename: {fl['filename']} -> uri: {fl['uri']}" for fl in file_links
+            )
+        )
+
+
+@nox.session(name="docs:links:check", python=False)
+def docs_links_check(session: Session) -> None:
+    """Checks whether all links in the documentation are accessible."""
+    ignore = [r"https?://"]
```

Comment:
Can you explain why we'd want to ignore links from https? afaics, locally, I wasn't having issues, unless the code was run multiple times, which led to rate limiting.